hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java

```diff
@@ -1698,7 +1701,19 @@ public boolean isEmpty() {
     }
 
     @Override
-    public void putAll(@Nonnull Map<? extends K, ? extends V> map) {
+    public void putAll(@Nonnull Map<? extends K, ? extends V> m) {
+        putAllInternal(m, null);
+    }
+
+    // used by Jet
+    public ICompletableFuture<Void> putAllAsync(@Nonnull Map<? extends K, ? extends V> m) {
+        SimpleCompletableFuture<Void> future = new SimpleCompletableFuture<>(ForkJoinPool.commonPool(),
```

Comment:
Currently client-side `ICompletableFuture` implementations returned to users use the `getClient().getClientExecutionService().getUserExecutor()` as default callback executor.